### ISSUE TITLE
[API+UI] Trigger checkpointing manually

### ISSUE THREAD

**Issue by bghira** [bghira/SimpleTuner#1816]
Similar vein to #1815 but we want to trigger checkpoint export in case of emergency, or for automated purposes via external tools.

The checkpoint would trigger while the training loop is going, after the next gradient sync completes.

This will allow the current gradient accumulation (if any) to complete first.
